app/models/issue.rb

```diff
@@ -16,8 +16,9 @@ def valid_for_user?(user, skip_update = Rails.env.test?)
       update_issue!
       return false if commenting_users.include?(user.github)
     end
-    return false  if closed?
-    return false  if pr_attached? && user.skip_issues_with_pr?
+    return false if closed?
+    return false if created_by?(user) && user.skip_my_own_issues_and_prs?
+    return false if pr_attached? && user.skip_issues_with_pr?
     true
```

Comment:
@O-I All the `return false` statements can be be extracted into a separate method.